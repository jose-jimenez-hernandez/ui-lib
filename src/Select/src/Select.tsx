import { Fragment, useState } from 'react';
import { Listbox, Transition } from '@headlessui/react';
import { ChevronDownIcon, CheckIcon, SelectorIcon } from '@heroicons/react/solid';
import { SelectProps } from './SelectType';
import useStyles from './styles';
import { Status } from '../../utils/types/StatusType';
import { StatusMessage } from '../StatusMessage';
import SelectOption from './SelectOption';

const Select = ({
  label,
  labelHint,
  status,
  statusMessage,
  hintMessage,
  disabled,
  placeholder,
  className,
  children,
  iconChevron,
  selectedItem,
}: Partial<SelectProps>) => {
  const [selected, setSelected] = useState(selectedItem);
  const hasError = status === Status.ERROR;
  const hasSuccess = status === Status.SUCCESS;
  const isChevronType = iconChevron !== 'selector';

  const styles = useStyles({
    className,
    disabled,
    error: hasError,
    success: hasSuccess,
  });

  return (
    <Listbox value={selected} onChange={setSelected}>
      {({ open }) => (
        <>
          <div className={styles.labelContainer}>
            {label && <Listbox.Label className={styles.label}>{label}</Listbox.Label>}
            {labelHint && <Listbox.Label className={styles.labelHint}>{labelHint}</Listbox.Label>}
          </div>
          <div className="mt-1 relative">
            <Listbox.Button className={styles.buttonBox.container}>
              <span className={styles.buttonBox.label}>{selected?.name || placeholder}</span>
              <span className={styles.buttonBox.chevron}>
                {isChevronType ? (
                  <ChevronDownIcon className={styles.buttonBox.chevronIcon} aria-hidden="true" />
                ) : (
                  <SelectorIcon className={styles.buttonBox.chevronIcon} aria-hidden="true" />
                )}
              </span>
            </Listbox.Button>

            <Transition
              show={open}
              as={Fragment}
              leave="transition ease-in duration-100"
              leaveFrom="opacity-100"
              leaveTo="opacity-0"
            >
              <Listbox.Options className={styles.options.container}>{children}</Listbox.Options>
            </Transition>
          </div>
          <StatusMessage status={status} defaultMessage={hintMessage} message={statusMessage} />
        </>
      )}
    </Listbox>
  );
};

export default Select;
