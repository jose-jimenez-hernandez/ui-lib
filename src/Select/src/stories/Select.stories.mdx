import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';

| Category   | Type    | Title  |
| ---------- | ------- | ------ |
| Components | General | Select |

Select component to select value from options.

## When To Use

- A dropdown menu for displaying choices - an elegant alternative to the native `<select>` element.


## API

| Property | Description                                           | Type                                   | Default    | 
| -------- | ----------------------------------------------------- | -------------------------------------- | ---------- | 
| disabled | Whether the select menu is disabled | boolean | - |
| placeholder | Placeholder of select | ReactNode | - |
| status          | Status of input                                | 'success' , 'error',  'none'           | 'none'    |
| statusMessage   | Setatus messages                               | string                                 | '' |  
| hintMessage     | Exta information about the field               | strin                                  | ''        |
| label           | Field primary label name                       | string                                 | ''        |
| labelHint       | Field secondary label name                     | string                                 | ''        |

It accepts all props which native support.
