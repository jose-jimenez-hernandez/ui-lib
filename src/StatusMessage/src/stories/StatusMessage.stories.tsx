import { Story } from '@storybook/react';
import { Status } from '../../../utils/types/StatusType';
import StatusMessage from '../StatusMessage';
import StatusMessageDoc from './StatusMessage.stories.mdx';

export default {
  title: 'Components/StatusMessage',
  component: StatusMessage,
  parameters: {
    controls: { exclude: /.*/, hideNoControlsWarning: true },
    docs: {
      page: (StatusMessageDoc as any).parameters?.docs.page,
    },
  },
};

const StatusMessageTemplate: Story = (args: any) => <StatusMessage {...args} />;

export const $PlayWithMe = StatusMessageTemplate.bind({});
$PlayWithMe.parameters = { controls: { exclude: null } };
$PlayWithMe.args = {
  defaultMessage: 'Default status message',
};

const StateTemplate: Story = () => (
  <>
    <StatusMessage defaultMessage="Default status message" status={Status.ERROR} message="Status error message" />
    <StatusMessage defaultMessage="Default status message" status={Status.SUCCESS} message="Status success message" />
    <StatusMessage defaultMessage="Default status message" status={Status.INFO} message="Status info message" />
    <StatusMessage defaultMessage="Default status message" status={Status.DEFAULT} message="Status default message" />
  </>
);

export const States = StateTemplate.bind({});
