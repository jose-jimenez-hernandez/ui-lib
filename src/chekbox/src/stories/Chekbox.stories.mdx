import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';

| Category   | Type    | Title  |
| ---------- | ------- | ------ |
| Components | General | Chekbox |

To trigger an operation.

## When To Use



## API

| Property | Description                                           | Type                                   | Default    | 
| -------- | ----------------------------------------------------- | -------------------------------------- | ---------- | 

It accepts all props which native support.
