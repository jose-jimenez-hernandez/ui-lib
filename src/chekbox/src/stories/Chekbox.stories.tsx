import React from 'react';
import { Meta } from '@storybook/react/types-6-0';
import { Story } from '@storybook/react';
import Chekbox from '../Chekbox';
import ChekboxDoc from './Chekbox.stories.mdx';

export default {
  title: 'Components/Chekbox',
  component: Chekbox,
  parameters: {
    docs: {
      page: (ChekboxDoc as any).parameters?.docs.page,
    },
  },
};

const ChekboxTemplate: Story = (args: any) => ( <Chekbox  {...args}/>
);

export const DefaultChekbox = ChekboxTemplate.bind({});
DefaultChekbox.args = {  };
