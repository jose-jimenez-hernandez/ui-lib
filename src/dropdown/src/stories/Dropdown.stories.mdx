import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';

| Category   | Type    | Title  |
| ---------- | ------- | ------ |
| Components | General | Dropdown |

To trigger an operation.

## When To Use



## API

| Property | Description                                           | Type                                   | Default    | 
| -------- | ----------------------------------------------------- | -------------------------------------- | ---------- | 

It accepts all props which native support.
