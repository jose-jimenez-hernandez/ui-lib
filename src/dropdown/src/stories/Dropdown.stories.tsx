import React from 'react';
import { Meta } from '@storybook/react/types-6-0';
import { Story } from '@storybook/react';
import Dropdown from '../Dropdown';
import DropdownDoc from './Dropdown.stories.mdx';
import DropdownItem from '../DropdownItem';

const items = [
  { id: '1', name: 'Wade Cooper' },
  { id: '2', name: 'Arlene Mccoy' },
  { id: '3', name: 'Devon Webb' },
  { id: '4', name: 'Tom Cook' },
  { id: '5', name: 'Tanya Fox' },
  { id: '6', name: 'Hellen Schmidt' },
  { id: '7', name: 'Caroline Schultz' },
  { id: '8', name: 'Mason Heaney' },
  { id: '9', name: 'Claudie Smitham' },
  { id: '10', name: 'Emil Schaefer' },
];

export default {
  title: 'Components/Dropdown',
  component: Dropdown,
  parameters: {
    docs: {
      page: (DropdownDoc as any).parameters?.docs.page,
    },
  },
};

const DropdownTemplate: Story = (args: any) => (
  <div className="flex justify-end">
    <Dropdown {...args}>
      {items &&
        items.map((item) => (
          <DropdownItem key={item.id} label={item.name} separator={true} value={item}></DropdownItem>
        ))}
    </Dropdown>
  </div>
);

export const DefaultDropdown = DropdownTemplate.bind({});
DefaultDropdown.args = {};
