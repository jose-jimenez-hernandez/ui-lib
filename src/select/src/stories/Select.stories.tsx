import { Story } from '@storybook/react';
import Select from '../Select';
import SelectOption from '../SelectOption';
import SelectDoc from './Select.stories.mdx';

const items = [
  { id: '1', name: 'Wade Cooper' },
  { id: '2', name: 'Arlene Mccoy' },
  { id: '3', name: 'Devon Webb' },
  { id: '4', name: 'Tom Cook' },
  { id: '5', name: 'Tanya Fox' },
  { id: '6', name: 'Hellen Schmidt' },
  { id: '7', name: 'Caroline Schultz' },
  { id: '8', name: 'Mason Heaney' },
  { id: '9', name: 'Claudie Smitham' },
  { id: '10', name: 'Emil Schaefer' },
];

export default {
  title: 'Components/Select',
  component: Select,
  parameters: {
    docs: {
      page: (SelectDoc as any).parameters?.docs.page,
    },
  },
};

const SelectTemplate: Story = (args: any) => (
  <Select {...args}>
    {items && items.map((item) => <SelectOption key={item.id} label={item.name} value={item}></SelectOption>)}
  </Select>
);

export const DefaultSelect = SelectTemplate.bind({});
DefaultSelect.args = { placeholder: 'Items ', label: 'Hello' };
