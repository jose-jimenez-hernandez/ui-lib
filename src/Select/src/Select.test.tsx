import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { composeStories } from '@storybook/testing-react';
import * as stories from './stories/Select.stories';

const { DefaultSelect } = composeStories(stories);

it('renders correctly', () => {
  const state = render(<DefaultSelect />).baseElement;

  expect(state).toMatchSnapshot();
});
