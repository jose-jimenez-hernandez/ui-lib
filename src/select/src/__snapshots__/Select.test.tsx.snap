// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<body>
  <div>
    <div
      class="flex justify-between"
    >
      <label
        class="block text-sm font-medium text-gray-700"
        id="headlessui-listbox-label-1"
      >
        Hello
      </label>
    </div>
    <div
      class="mt-1 relative"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        aria-labelledby="headlessui-listbox-label-1 headlessui-listbox-button-undefined"
        class="bg-white relative w-full border rounded-md shadow-sm pl-3 pr-10 py-2 text-left cursor-default focus:outline-none focus:ring-1 focus:ring-primary-500 focus:border-primary-500 sm:text-sm border-gray-300"
        id="headlessui-listbox-button-2"
        type="button"
      >
        <span
          class="block truncate text-gray-500"
        >
          Items 
        </span>
        <span
          class="absolute inset-y-0 right-0 flex items-center pr-2 pointer-events-none"
        >
          <svg
            aria-hidden="true"
            class="h-5 w-5 text-gray-400"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</body>
`;
